unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs

```diff
@@ -0,0 +1,226 @@
+        [SerializeField] internal Transform activeQuestsContainer;
+        [SerializeField] internal ActiveQuestComponentView activeQuestPrefab;
+
+        public event Action<string, bool> OnPinChange;
+        public event Action<Vector2Int> OnJumpIn;
+        public event Action<string> OnQuestAbandon;
+
+        private Dictionary<string, ActiveQuestComponentView> activeQuests;
+        private Dictionary<string, ActiveQuestComponentView> completedQuests;
+        private UnityObjectPool<ActiveQuestComponentView> questsPool;
+        private string previouslyActiveSelectedQuest;
+        private string previouslyCompletedSelectedQuest;
+
+        public override void Awake()
+        {
+            activeQuests = new ();
+            completedQuests = new ();
+            questsPool = new UnityObjectPool<ActiveQuestComponentView>(activeQuestPrefab, activeQuestsContainer, actionOnDestroy: x => x.Hide());
+            questsPool.Prewarm(MAX_QUESTS_COUNT);
+
+            InitialiseSectionSelector();
+
+            questDetailsComponentView.OnJumpIn += (coords) => OnJumpIn?.Invoke(coords);
+            questDetailsComponentView.OnPinChange += (questId, isPinned) => OnPinChange?.Invoke(questId, isPinned);
+            questDetailsComponentView.OnQuestAbandon += (questId) => OnQuestAbandon?.Invoke(questId);
+
+            emptyState.SetActive(true);
+        }
+
+        private void InitialiseSectionSelector()
+        {
+            sectionSelector.Awake();
+            sectionSelector.GetSection(IN_PROGRESS_SECTION_INDEX).onSelect.RemoveAllListeners();
+            sectionSelector.GetSection(COMPLETED_SECTION_INDEX).onSelect.RemoveAllListeners();
+            sectionSelector.GetSection(IN_PROGRESS_SECTION_INDEX).onSelect.AddListener((isSelected) =>
+            {
+                if (!isSelected) return;
+
+                headerText.text = IN_PROGRESS_TITLE;
+                ShowActiveOrCompletedQuests(true);
+                if (activeQuests.Count == 0)
+                {
+                    emptyState.SetActive(true);
+                    emptyActiveState.SetActive(true);
+                    emptyCompletedState.SetActive(false);
+                }
+                else
+                {
+                    emptyState.SetActive(false);
+                }
+            });
+            sectionSelector.GetSection(COMPLETED_SECTION_INDEX).onSelect.AddListener((isSelected) =>
+            {
+                if (!isSelected) return;
+
+                headerText.text = COMPLETED_TITLE;
+                ShowActiveOrCompletedQuests(false);
+                if (completedQuests.Count == 0)
+                {
+                    emptyState.SetActive(true);
+                    emptyActiveState.SetActive(false);
+                    emptyCompletedState.SetActive(true);
+                }
+                else
+                {
+                    emptyState.SetActive(false);
+                }
+            });
```

Comment:
Can we extract those anons? just for clarity. `InProgressSectionOnSelect` and `CompletedSectionOnSelect` or whatever name fits 😄